test/gtest/CMakeLists.txt

```diff
@@ -7,9 +7,13 @@ include(GoogleTest)
 
 find_package(rocblas)
 
+set(SOURCES ${SOURCES}
+    "${CMAKE_CURRENT_SOURCE_DIR}/log_test_helper.cpp"
+    )
+
 function(add_gtest TEST_NAME)
   message("Adding Test: " ${TEST_NAME})
-  add_executable(test_${TEST_NAME} ${TEST_NAME}.cpp )
+  add_executable(test_${TEST_NAME} ${TEST_NAME}.cpp ${SOURCES})
```

Comment:
This adds SOURCES to the list of sources for _each_ test, which is wrong. `log_test_helper` is necessary for `log_test*` executables only.

Reply:
Actually I see several problems related to the notion of "helpers":
- necessity to manually maintain the list of helpers
- necessity to remove then from TESTS
- extra dependencies

[Recommendation] Currently the simplest solution to me is to merge `log_test` and `log_test_neg` into a single .cpp and get rid of helpers entirely.

Reply:
In MIOpen we can only `set` environment variable once during a life time of a process.

MIOpen uses IsEnabled function ([env.hpp](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/include/miopen/env.hpp#)) to query environment variable.

See the function defition below :

 
`template <class T>
inline bool IsEnabled(T)
{
    static const bool result = miopen::IsEnvvarValueEnabled(T::value());
    return result;
}`

Since the initialization of static objects is performed only once. `IsEnabled` will always return the value `result` was initialized with.

This is why we need two files to test
1) start with setting environment variable. 

2) start with no environment variable set at all. Test the negative case.

Reply:
We could also add a new folder `helper` that would contain all the helper `*.cpp` eg : `log_test_helper.cpp`. In the CMakeListst.txt 
`file(GLOB CPP_HELPER helper/*.cpp)`. Then add this variable to add_executable
eg: 
` add_executable(test_\${TEST_NAME} ${TEST_NAME}.cpp ${CPP_HELPER})`

Reply:
@bghimireamd Thanks for explanation. The idea of the helper folder looks good. If the implementation of helper-folder turns out to be too complicated you can use another approach, e.g. use the same prefix for all tests (or for all helpers).